src/transactionsPage/utils.ts

```diff
@@ -0,0 +1,120 @@
+import { Transaction, Filters } from "./";
+import { SelectOptions } from "./filter";
+import { StatementStatistics } from "src/util/appStats";
+
+export const getAppNames = (transactions: Transaction[]) => {
+  return transactions.reduce(
+    (acc, current) => {
+      const twin = acc.some(
+        (item: SelectOptions) => item.label === current.stats_data.app,
+      );
+      return twin
+        ? acc
+        : acc.concat([
+            {
+              label: current.stats_data.app,
+              value: current.stats_data.app,
+            },
+          ]);
+    },
+    [{ label: "All", value: "All" }],
+  );
+};
+
+export const addTransactionStatements = (
+  transactions: Transaction[],
+  statements: StatementStatistics[],
+) => {
+  const data = transactions.map((item: Transaction) => {
+    const transactionStatements = item.stats_data.statement_ids.reduce(
+      (acc: string, current: string) => {
+        const statement = statements.find(
+          (stItem: StatementStatistics) => stItem.key.key_data.id === current,
+        );
+        // const newLine = idx > 0 ? "\n" : "";
+        return statement ? `${acc} ${statement.key.key_data.query}` : acc;
+      },
+      "",
+    );
+    return {
+      transactionStatements,
+      ...item,
+    };
```

Comment:
In general, I would avoid doing this and create a new type that contains a `Transaction` and a `string` that's the transaction's concat-ed statement. This creates a clear boundary between server-side data and computed data on the client-side.